aiida/tools/importexport/dbimport/backends/django.py

```diff
@@ -0,0 +1,689 @@
+                    progress.update()
+                    # Note: UUIDs need to be converted to strings
+                    relevant_db_entries.update({str(getattr(object_, unique_identifier)): object_})
+
+    foreign_ids_reverse_mappings[entity_name] = {k: v.pk for k, v in relevant_db_entries.items()}
+
+    entity_count = reader.entity_count(entity_name)
+    if not entity_count:
+        return
+
+    with get_progress_reporter()(desc=f'Reading archived entities - {entity_name}', total=entity_count) as progress:
+        imported_comp_names = set()
+        for pk, fields in reader.iter_entity_fields(entity_name):
+            progress.update()
+            if entity_name == GROUP_ENTITY_NAME:
+                # Check if there is already a group with the same name
+                dupl_counter = 0
+                orig_label = fields['label']
+                while model.objects.filter(label=fields['label']):
+                    fields['label'] = orig_label + DUPL_SUFFIX.format(dupl_counter)
+                    dupl_counter += 1
+                    if dupl_counter == 100:
+                        raise exceptions.ImportUniquenessError(
+                            f'A group of that label ( {orig_label} ) already exists and I could not create a new one'
+                        )
+
+            elif entity_name == COMPUTER_ENTITY_NAME:
+                # Check if there is already a computer with the same name in the database
+                dupl = (model.objects.filter(name=fields['name']) or fields['name'] in imported_comp_names)
+                orig_name = fields['name']
+                dupl_counter = 0
+                while dupl:
+                    # Rename the new computer
+                    fields['name'] = orig_name + DUPL_SUFFIX.format(dupl_counter)
+                    dupl = (model.objects.filter(name=fields['name']) or fields['name'] in imported_comp_names)
+                    dupl_counter += 1
+                    if dupl_counter == 100:
+                        raise exceptions.ImportUniquenessError(
+                            f'A computer of that name ( {orig_name} ) already exists and I could not create a new one'
+                        )
+
+                imported_comp_names.add(fields['name'])
+
+            if fields[unique_identifier] in relevant_db_entries:
+                # Already in DB
+                existing_entries[entity_name][str(pk)] = fields
+            else:
+                # To be added
+                if entity_name == NODE_ENTITY_NAME:
+                    # format extras
+                    fields = _sanitize_extras(fields)
+                    if extras_mode_new != 'import':
+                        fields.pop('extras', None)
+                new_entries[entity_name][str(pk)] = fields
+
+
+def _store_entity_data(
+    *, reader: ArchiveReaderAbstract, entity_name: str, comment_mode: str, extras_mode_existing: str,
```

Comment:
IMO you should always use a `*` before keyword arguments, i.e. enforcing the use of `func(a=1, b=2)` as opposed to `func(1, 2)` (and this should be implemented more in aiida). It makes the function more resistant to refactoring, because calls to the function will not accidentally retrieve the wrong arguments if they have been re-ordered or removed.